components/Posts/PostsPreview.tsx

```diff
@@ -1,7 +1,18 @@
-import { Card } from '@sanity/ui'
+import { Card, Text } from '@sanity/ui'
 import React from 'react'
 
 export function PostsPreview(props: any) {
+  // if the document has no slug for the preview iframe
+  if (!props.document.displayed.slug) {
+    return (
+      <Card tone="primary" margin={5} padding={6}>
+        <Text align="center">
+          Please add a slug to the post to see the preview!
+        </Text>
+      </Card>
```

Comment:
@mikolajdobrucki I've added a visual aid when the preview cannot show content (when the slug is not set). Let me know if you want to make any changes / make it better or if this is fine :) 

<img width="1385" alt="image" src="https://user-images.githubusercontent.com/6951139/196412858-f4c6f6e4-a2ac-459b-acbe-23cf0c009c18.png">